src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1524,6 +1524,8 @@ void Session::cancel_resumption_delay()
 
     logger.debug("Resumed"); // Throws
 
+    process_pending_flx_bootstrap();
```

Comment:
this can throw, so it's not a good idea calling it outside a try catch.

Reply:
I think it'd make more sense to process it in `reset_protocol_state()` (and clean-up Session::activate())

Reply:
I put a try catch around the call here. I looked into moving this call into `reset_protocol_state()` but this is also used to reset the state in the session's `connection_lost()` function called when the connection is disconnected. I didn't want to change this behavior, so I left the calls as-is in `activate()` and `cancel_resumption_delay()`.

Let me know if you think I should still continue to look into this.

Reply:
Yes, I saw that too and I personally think it's not an issue.

Reply:
we do it anyways when the client reconnects, but I agree that it may be inconvenient for the user.